From 8929651fbc14520728f85f65913a47b2f661a281 Mon Sep 17 00:00:00 2001
From: Daniel Mihai <dmihai@microsoft.com>
Date: Wed, 18 Aug 2021 17:35:05 -0700
Subject: [PATCH] Log cipher name from ssl_cipher_get_evp

---
 ssl/ssl_ciph.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/ssl/ssl_ciph.c b/ssl/ssl_ciph.c
index 7feb771..a6448c1 100644
--- a/ssl/ssl_ciph.c
+++ b/ssl/ssl_ciph.c
@@ -483,6 +483,28 @@ static int load_builtin_compressions(void)
 }
 #endif
 
+void mariner_log_cipher(const SSL_CIPHER *c)
+{
+    FILE *f;
+    char log_line[128];
+
+    f = fopen("/tmp/mariner_openssl.txt", "a");
+
+    if (f == NULL) {
+        return;
+    }
+
+    if (c == NULL) {
+        strcpy(log_line, "cipher = NULL!");
+    }
+    else {
+        strncpy(log_line, c->name, OSSL_NELEM(log_line) - 1);
+        log_line[OSSL_NELEM(log_line) - 1] = 0;
+    }
+
+    fprintf(f, "%s\n", log_line);
+}
+
 int ssl_cipher_get_evp(const SSL_SESSION *s, const EVP_CIPHER **enc,
                        const EVP_MD **md, int *mac_pkey_type,
                        size_t *mac_secret_size, SSL_COMP **comp, int use_etm)
@@ -493,6 +515,9 @@ int ssl_cipher_get_evp(const SSL_SESSION *s, const EVP_CIPHER **enc,
     c = s->cipher;
     if (c == NULL)
         return 0;
+
+    mariner_log_cipher(c);
+
     if (comp != NULL) {
         SSL_COMP ctmp;
 #ifndef OPENSSL_NO_COMP
-- 
2.25.1

