--- CBL-Mariner-Linux-Kernel-rolling-lts-mariner-5.10.28.1/crypto/testmgr.h	2021-05-26 17:23:55.256900091 -0700
+++ CBL-Mariner-Linux-Kernel-rolling-lts-mariner-5.10.28.1/crypto/testmgr.h	2021-05-26 17:24:12.656864524 -0700
@@ -9104,7 +9104,7 @@
 	},
 };
 
-static const struct cipher_testvec des3_ede_cbc_tv_template[] = {
+static struct cipher_testvec des3_ede_cbc_tv_template[] = {
 	{ /* Generated from openssl */
 		.key	= "\xE9\xC0\xFF\x2E\x76\x0B\x64\x24"
 			  "\x44\x4D\x99\x5A\x12\xD6\x40\xC0"
