From 43ad9be22ce7e0599eea67dab299e85d6333dff4 Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Thu, 9 Feb 2023 14:37:21 +0000
Subject: [PATCH 25/31] usermanage: Handle symlinks in /usr/share/cracklib.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/admin/usermanage.if | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/admin/usermanage.if b/policy/modules/admin/usermanage.if
index 340d02456..75589e353 100644
--- a/policy/modules/admin/usermanage.if
+++ b/policy/modules/admin/usermanage.if
@@ -322,4 +322,5 @@ interface(`usermanage_read_crack_db',`
 
 	files_search_var($1)
 	read_files_pattern($1, crack_db_t, crack_db_t)
+	read_lnk_files_pattern($1, crack_db_t, crack_db_t)
 ')
-- 
2.25.1

